### DISCUSSION TITLE
Memory management for 2 input processor.

### DISCUSSION THREAD

**Discussion by Bauaser-kun** [apache/streampipes#2334] [Q&A]
Hi I've created a my first processor. The idea is pretty simple - take 2 input streams (total production & waste count) and output waste percentage from a batch.
THe processor uses the same stream buffer as OOTB merge 2 streams processor but as I can see the buffer keeps increaing in size with each event.
Is there any in-build memory management in event handler to not run out of memory? And if not is there any preffered way to efficiently clean buffer after an event/specified time?

**Comment by tenthe** [comment]
Hello @Bauaser-kun,
thank you for the question. That is a good point. There is no built-in memory management.
The preferred way of handling state in memory depends on the use case.
In general, you should ensure that an appropriate cleanup strategy is in place.
However, since this is a common case, it would also be good to have a generic service that takes care of the state.
If you have a suggestion on how such a service could look like, post it here and we can discuss it.
What do you think of it?

**Reply by Bauaser-kun** [reply]
Sorry for the late response. As of know I don't have any suggestions, as for now we (as a company) are checking possibilites and more like playing with streampipes. We will however propably implement them internally and will discuss memory usage in bigger team.

**Comment by tenthe** [comment]
Thank you for your feedback. I would close the discussion. If you have any questions, please feel free to open it again. 
Also if there are any other questions please do not hesitate to contact us.
